execution/types/runtime_decoder_test.go

```diff
@@ -0,0 +1,344 @@
+		// extra
+		uint8(moduleOneIdx), uint8(functionIdx), // call
+	}
+
+	header = primitives.Header{
+		ParentHash: primitives.Blake2bHash{
+			FixedSequence: sc.BytesToFixedSequenceU8(parentHash)},
+		Number:         5,
+		StateRoot:      primitives.H256{FixedSequence: sc.BytesToFixedSequenceU8(stateRoot)},
+		ExtrinsicsRoot: primitives.H256{FixedSequence: sc.BytesToFixedSequenceU8(extrinsicsRoot)},
+		Digest:         primitives.Digest{},
+	}
+
+	parentHash      = common.MustHexToHash("0x3aa96b0149b6ca3688878bdbd19464448624136398e3ce45b9e755d3ab61355c").ToBytes()
+	stateRoot       = common.MustHexToHash("0x3aa96b0149b6ca3688878bdbd19464448624136398e3ce45b9e755d3ab61355b").ToBytes()
+	extrinsicsRoot  = common.MustHexToHash("0x3aa96b0149b6ca3688878bdbd19464448624136398e3ce45b9e755d3ab61355a").ToBytes()
+	moduleFunctions = map[sc.U8]primitives.Call{}
+)
+
+var (
+	mockModuleOne *mocks.Module
+	mockCallOne   *mocks.Call
+)
+
+func Test_RuntimeDecoder_New(t *testing.T) {
+	target := setupRuntimeDecoder()
+	expect := runtimeDecoder{
+		modules: []primitives.Module{mockModuleOne},
+		extra:   mockSignedExtra,
+	}
+
+	assert.Equal(t, expect, target)
+}
+
+func Test_RuntimeDecoder_DecodeBlock_ZeroExtrinsicsEmptyBody(t *testing.T) {
+	target := setupRuntimeDecoder()
+
+	lenExtrinsics := sc.ToCompact(0).Bytes()
+	buff := bytes.NewBuffer(append(header.Bytes(), lenExtrinsics...))
+
+	resultBlock := target.DecodeBlock(buff)
+
+	expectedBlock := NewBlock(header, sc.Sequence[primitives.UncheckedExtrinsic]{})
+
+	assert.Equal(t, expectedBlock, resultBlock)
+}
+
+func Test_RuntimeDecoder_DecodeBlock_Module_Not_Exists(t *testing.T) {
+	target := setupRuntimeDecoder()
+
+	idxModuleNotExists := sc.U8(10)
+
+	nonExistentExtrinsicsBytes := []byte{
+		byte(signedExtrinsicVersion),                                                                      // version
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // signer
+		0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+		1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+		1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+		1, 1, 1, 1, 1, 1, 1, // signature,
+		// extra
+		uint8(idxModuleNotExists), uint8(functionIdx), // call
+	}
+
+	buffExtrinsic := bytes.NewBuffer(append(sc.ToCompact(len(nonExistentExtrinsicsBytes)).Bytes(), nonExistentExtrinsicsBytes...))
+
+	lenExtrinsics := sc.ToCompact(1).Bytes()
+	decodeBlockBytes := append(lenExtrinsics, buffExtrinsic.Bytes()...)
+	decodeBlockBytes = append(header.Bytes(), decodeBlockBytes...)
+
+	decodeBlockBuff := bytes.NewBuffer(decodeBlockBytes)
+
+	mockModuleOne.On("GetIndex").Return(moduleOneIdx)
+	mockSignedExtra.On("Decode", mock.Anything).Return()
+
+	assert.PanicsWithValue(t, "module with index ["+strconv.Itoa(int(idxModuleNotExists))+"] not found", func() {
+		target.DecodeBlock(decodeBlockBuff)
+	})
+}
+
+func Test_RuntimeDecoder_DecodeBlock_Function_Not_Exists(t *testing.T) {
```

Comment:
I think that this test has to be for DecodeCall.

Reply:
Fixed